src/schemas/set.schema.ts

```diff
@@ -0,0 +1,26 @@
+import { Set } from "@/models/set.model";
+import mongoose, { Schema } from "mongoose";
+
+const setSchema = new Schema<Set>({
+  setNumber: {
+    type: Number,
+    required: true,
+    unique: true,
+  },
+  weight: {
+    type: Number,
+    required: true,
+  },
+  unit: {
+    type: String,
+    required: true,
+  },
+  reps: {
+    type: Number,
+    required: true,
+  }
+})
+
+
+export const SetRepository: mongoose.Model<Set> =
+  mongoose.models.Set || mongoose.model("Set", setSchema);
```

Comment:
Same here. No need to create and export this object. Export the setSchema and use it in the other schema where we need it.